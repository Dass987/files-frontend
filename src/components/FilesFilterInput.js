import React from 'react';
import Form from 'react-bootstrap/Form';
import { useDispatch } from 'react-redux';

import { fetchFiles } from '../features/files/filesSlice';

export const FilesFilterInput = () => {
  const dispatch = useDispatch();

  const filterByName = e => {
    dispatch(fetchFiles(e.target.value));
  };

  return (
    <div className="mb-3">
      <Form.Control size="sm" placeholder="Search by file name..." aria-label="Search" aria-describedby="basic-addon1" onChange={filterByName} />
    </div>
  );
};
