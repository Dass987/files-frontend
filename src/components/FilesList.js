import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import Table from 'react-bootstrap/Table';
import { fetchFiles } from '../features/files/filesSlice';

export const FilesList = () => {
  const filesState = useSelector(state => state.files);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchFiles());
  }, [dispatch]);

  return (
    <Table responsive striped bordered style={{ borderColor: '#969696' }}>
      <thead>
        <tr style={{ borderBottom: '1px solid gray' }}>
          <th>File Name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {filesState.rows.length > 0 ? (
          filesState.rows.map((row, i) =>
            row.lines.map((line, j) => (
              <tr key={`${i}-${row.file}-${j}`}>
                <td>{row.file}</td>
                <td>{line.text}</td>
                <td>{line.number}</td>
                <td>{line.hex}</td>
              </tr>
            ))
          )
        ) : (
          <tr>
            <td colSpan={4} className="text-center py-4">
              {filesState.status === 'pending' ? 'Loading...' : 'No results'}
            </td>
          </tr>
        )}
      </tbody>
    </Table>
  );
};
