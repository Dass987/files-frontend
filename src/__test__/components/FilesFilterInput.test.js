import React from 'react';
import { render, screen } from '@testing-library/react';

import { FilesFilterInput } from '../../components/FilesFilterInput';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

test('renders input element', () => {
  render(<FilesFilterInput />);

  const input = screen.getByRole('textbox');
  expect(input).toBeInTheDocument();
});

test('input has correct attributes', () => {
  render(<FilesFilterInput />);

  const input = screen.getByRole('textbox');
  expect(input).toHaveAttribute('aria-label', 'Search');
  expect(input).toHaveAttribute('placeholder', 'Search by file name...');
});

test('renders with expected output', () => {
  const { asFragment } = render(<FilesFilterInput />);
  expect(asFragment()).toMatchSnapshot();
});
