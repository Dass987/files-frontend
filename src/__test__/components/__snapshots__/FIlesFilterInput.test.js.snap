// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with expected output 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <input
      aria-describedby="basic-addon1"
      aria-label="Search"
      class="form-control form-control-sm"
      placeholder="Search by file name..."
    />
  </div>
</DocumentFragment>
`;
